src/EventStore.Core/Util/StringFilter.cs

```diff
@@ -88,5 +114,54 @@ public bool IsStringAllowed(string s)
                 return _allowedStringsRegex.IsMatch(s);
             }
         }
+
+        private class RegexCollectionStrategy : IFilterStrategy
+        {
+            private readonly List<Regex> regexes;
+
+            public RegexCollectionStrategy(List<Regex> regexes)
+            {
+                this.regexes = regexes;
+            }
+
+            public static bool IsRegexString(string s)
+            {
+                return !s.StartsWith("$", System.StringComparison.Ordinal) && !s.Equals(Regex.Escape(s));
+            }
+
+            public bool IsStringAllowed(string s)
+            {
+                foreach(var regex in regexes)
+                {
+                    if(regex.IsMatch(s))
+                    {
+                        return true;
+                    }
+                }
+                return false;
+            }
+        }
+
+        private class MultiStrategyStrategy : IFilterStrategy
+        {
+            private readonly List<IFilterStrategy> strategies;
+
+            public MultiStrategyStrategy(List<IFilterStrategy> strategies)
+            {
+                this.strategies = strategies;
+            }
+
+            public bool IsStringAllowed(string s)
+            {
+                foreach(var strat in strategies)
```

Comment:
First impression is that this would make the code less clear. `Exists` sounds to me like we expect something to exist. But it is just a filter. I am probably not c# developer enough to like this.

Reply:
I am fine with both. For me Exists looks c#y

Reply:
Or use Any, which is an extension method for Enumerables.
``` csharp
    public static bool Any<TSource>(this IEnumerable<TSource> source, Func<TSource, bool> predicate)
    {
      if (source == null)
        throw Error.ArgumentNull(nameof (source));
      if (predicate == null)
        throw Error.ArgumentNull(nameof (predicate));
      foreach (TSource source1 in source)
      {
        if (predicate(source1))
          return true;
      }
      return false;
    }
```

Reply:
Or just leave it as is!

Reply:
I'll leave it as is then.